sdk/src/types/block/semantic.rs

```diff
@@ -209,294 +210,279 @@ impl<'a> ValidationContext<'a> {
+                Output::Basic(output) => (
+                    output.unlock(output_id, unlock, &mut self),
+                    output.amount(),
+                    output.mana(),
+                    Some(output.native_tokens()),
+                    output.unlock_conditions(),
+                ),
+                Output::Account(output) => (
+                    output.unlock(output_id, unlock, &mut self),
+                    output.amount(),
+                    output.mana(),
+                    Some(output.native_tokens()),
+                    output.unlock_conditions(),
+                ),
+                Output::Foundry(output) => (
+                    output.unlock(output_id, unlock, &mut self),
+                    output.amount(),
+                    0,
+                    Some(output.native_tokens()),
+                    output.unlock_conditions(),
+                ),
+                Output::Nft(output) => (
+                    output.unlock(output_id, unlock, &mut self),
+                    output.amount(),
+                    output.mana(),
+                    Some(output.native_tokens()),
+                    output.unlock_conditions(),
+                ),
+                Output::Delegation(output) => (
+                    output.unlock(output_id, unlock, &mut self),
+                    output.amount(),
+                    0,
+                    None,
+                    output.unlock_conditions(),
+                ),
+            };
+
+            if let Err(conflict) = conflict {
+                return Ok(Some(conflict));
+            }
 
-        if unlock_conditions.is_time_locked(context.transaction.creation_slot()) {
-            return Ok(Some(TransactionFailureReason::TimelockNotExpired));
-        }
+            if unlock_conditions.is_time_locked(self.transaction.creation_slot()) {
+                return Ok(Some(TransactionFailureReason::TimelockNotExpired));
+            }
 
-        if !unlock_conditions.is_expired(context.transaction.creation_slot()) {
-            if let Some(storage_deposit_return) = unlock_conditions.storage_deposit_return() {
-                let amount = context
-                    .storage_deposit_returns
-                    .entry(storage_deposit_return.return_address().clone())
-                    .or_default();
+            if !unlock_conditions.is_expired(self.transaction.creation_slot()) {
+                if let Some(storage_deposit_return) = unlock_conditions.storage_deposit_return() {
+                    let amount = self
+                        .storage_deposit_returns
+                        .entry(storage_deposit_return.return_address().clone())
+                        .or_default();
 
-                *amount = amount
-                    .checked_add(storage_deposit_return.amount())
-                    .ok_or(Error::StorageDepositReturnOverflow)?;
+                    *amount = amount
```

Comment:
now that i see this again, are we sure this updates the correct amount variable?

Reply:
Looks good to me